### PR TITLE
Make disable_pypi_version_check configurable from client creation

### PR DESCRIPTION
**Pull request by mathialo**
Add an option to turn off the version check from an argument to init

### CODE REVIEW

File: cognite/client/_cognite_client.py [hunk 1]
```diff
@@ -48,6 +49,7 @@ def __init__(
         headers: Dict[str, str] = None,
         timeout: int = None,
         token: Union[str, Callable[[], str], None] = None,
+        disable_pypi_version_check: bool = None,
```
Comment (erlendvollset) [inline cognite/client/_cognite_client.py]:
```suggestion
        disable_pypi_version_check: bool = False,
```

Comment (mathialo) [reply cognite/client/_cognite_client.py]: I agree thats a bit clunky, but then how we detect no argument so we can fall back to env vars?

Comment (erlendvollset) [reply cognite/client/_cognite_client.py]: Ah right. Set type hint to Optional[bool] in that case.

File: cognite/client/utils/_client_config.py [hunk 2]
```diff
@@ -75,6 +76,7 @@ def __init__(
         self.headers = headers or self.headers
         self.timeout = timeout or self.timeout
         self.token = token
+        self.disable_pypi_version_check = disable_pypi_version_check or self.disable_pypi_version_check
```
Comment (erlendvollset) [inline cognite/client/utils/_client_config.py]:
```suggestion
        self.disable_pypi_version_check = disable_pypi_version_check if disable_pypi_version_check is not None else self.disable_pypi_version_check
```

File: cognite/client/utils/_client_config.py [hunk 3]
```diff
@@ -63,6 +63,7 @@ def __init__(
         headers: Dict[str, str] = None,
         timeout: int = None,
         token: Union[Callable[[], str], str] = None,
+        disable_pypi_version_check: bool = None,
```
Comment (erlendvollset) [inline cognite/client/utils/_client_config.py]: Optional[bool]

File: cognite/client/utils/_client_config.py (hunk 2 above)
Comment (erlendvollset) [reply cognite/client/utils/_client_config.py]: Will use envvar when this is false.

**Comment by sanderland** [comment]
my inclination is to call the parameter `check_version` with default True, rather than this double negative.

**Comment by mathialo** [comment]
> my inclination is to call the parameter `check_version` with default True, rather than this double negative.

I don't disagree (while on the topic of double negatives...), my reasoning was to keep it equal to the existing envvar-way of disabling it.

**Pull request merged**